### ISSUE TITLE
Further Clarifications

### ISSUE THREAD

**Issue by AleenaAzam** [dattran2346/xray-thesis#3]
Hi @dattran2346,
Can you please clarify a few questions:

1. What does the cut_all.pickle file contain?
2. For the training of the Unet model, the labels for the data is the binary image containing a mask of both lungs right?
3. Can you please guide us a bit more as to how to test your Unet model on another data set?
4. Also, if possible, can you give a brief description of what each of your .ipynb file is for?

@AleenaAzam

**Comment by dattran2346** [comment]
Hi @AleenaAzam,
1. As you can see in `src/Cropped Lung.ipynb`, the `cut_all.pickle` contains the lung bounding box in `left, top, right, bottom` format. 
2. Yes, I fused both the left and right lung in a single binary image.
3. Here is the template code that might help you to run the pretrained model on new dataset


```
tfm = transforms.Compose([
                transforms.Resize((256, 256)), # the model was trained on 256x256 image
                ... ])
                
# you may need to modify unet.py to load the pretrained model
unet = Unet(..)
unet.eval() # this line is critical

dataset = YourDataset(tfm, ...) 
dataloader = DataLoader(dataset, ...) 

for images, targets in dataloader:
	... 
	logits = torch.sigmoid(unet(image))
   predicts = logits > 0.5
	...
	
	# Run morphological opening to remove noise

```

Take a look at [these sample code](https://github.com/dattran2346/chestX-ray-14/blob/ca3ae780f2f0cc7d423549dd8cd90886ffc511fa/src/unet.py#L113-L136) 

4. I think that you are doing a project on Lung Segmentation, if so these notebooks might help you:

- `Cropped Lung.ipynb`: I run unet model trained on JSRT and MC in ChestXray14 and do post-processing step, i.e, morphological opening
- `Lung Segmentation.ipynb`: I trained the unet model on JSRT and MC in this notebook, and since segmentation is not my main task, I didnt train until convergence.
- `Predict.ipynb`: Run the chexnet model on cropped lung (after lung segmentation) and visualize the result
- `Preprocess Lung Segment.ipynb`: Prepare training data for lung segmentation, the fused binary mask is here
- `Test.ipynb`: Test one-stage (chexnet) and two-stage (unet + chexnet) model

**Comment by AleenaAzam** [comment]
@dattran2346 Thank you so much for all your help!
One more question, which version of fastai are you using: version 0.7, version 1 or are you using both somehow? I am having compatibility issues with fastai, torch and torchvision in some files, but not in others.

**Comment by dattran2346** [comment]
@AleenaAzam, I have included the the snapshot of fastai library that I used under `src/fastai`.
